File: src/main/java/ac/dnd/server/admission/infrastructure/persistence/entity/Event.java
package ac.dnd.server.admission.infrastructure.persistence.entity;

import java.time.LocalDateTime;

import ac.dnd.server.admission.domain.model.ViewablePeriod;
import ac.dnd.server.common.support.BaseEntity;
import ac.dnd.server.common.support.EventStatus;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "event")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Event extends BaseEntity {
	private String name;

	@Embedded
	private ViewablePeriod period;

	private LocalDateTime resultAnnouncementDateTime;

	@Enumerated(EnumType.STRING)
	private EventStatus status;

	public Event(
		final String name,
		final ViewablePeriod period,
		final LocalDateTime resultAnnouncementDateTime,
		final EventStatus status
	) {
		this.name = name;
		this.period = period;
		this.resultAnnouncementDateTime = resultAnnouncementDateTime;
		this.status = status;
	}

}

File: src/main/kotlin/ac/dnd/server/admission/infrastructure/persistence/entity/Event.kt
package ac.dnd.server.admission.infrastructure.persistence.entity

import ac.dnd.server.admission.domain.model.ViewablePeriod
import ac.dnd.server.global.base.BaseEntity
import ac.dnd.server.enums.EventStatus
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "event")
class Event(
    val name: String,

    @Embedded
    val period: ViewablePeriod,

    val resultAnnouncementDateTime: LocalDateTime,

    @Enumerated(EnumType.STRING)
    val status: EventStatus
) : BaseEntity() {

    protected constructor() : this(
        "",
        ViewablePeriod.of(LocalDateTime.now(), LocalDateTime.now()),
        LocalDateTime.now(),
        EventStatus.PENDING
    )
}
